yarn-project/noir-libs/noir-aztec/src/types/grumpkin_scalar.nr

```diff
@@ -0,0 +1,30 @@
+use crate::types::type_serialisation::TypeSerialisationInterface;
+
+struct GrumpkinScalar {
+    high: Field,
+    low: Field,
+}
+
+impl GrumpkinScalar {
+    fn new(high: Field, low: Field) -> Self {
+        GrumpkinScalar { high, low }
```

Comment:
Should we assert that `high` is below a value such that `high ++ low` is below the grumpkin scalar modulus?

Reply:
It would absolutely be a good idea to do a size check here. Do you know of some way how to do it without the sum overflowing the `Field` type and the high low values not being converted to a different type?

I did some research and it seems that the conversion to a larger type is the only feasible approach (but that's probably quite costly).

 If not, please don't spend much time on this.